src/noteCard/NoteCard.tsx

```diff
@@ -1,36 +1,159 @@
+
+  const { control, handleSubmit } = useForm<MockDataItem>({
+    defaultValues: item,
+  });
+  const [isEditing, setIsEditing] = useState<boolean>(true);
+
+  const onSubmit: SubmitHandler<MockDataItem> = (item) => {
+    updateBox(item);
+
+    setIsEditing((isEditing) => !isEditing);
+  };
+
+  // const modifedDimension = item.dimension
+  //   .map((dim) => dim + " x ")
+  //   .join("")
+  //   .slice(0, -2);
+
   return (
     <div>
-      <Card key={item._id}>
-        <CardHeader
-          action={
-            <IconButton onClick={() => useDelete(item._id)}>
-              <DeleteOutlineIcon />
-            </IconButton>
-          }
-          title={item.name}
-        />
-        <CardContent>
-          {/* <Typography variant="body2" color="textSecondary">
-            {item.dimension.map((val) => val + "  ")}
-          </Typography> */}
-          <Typography variant="body2" color="textSecondary">
-            {item.usage}
-          </Typography>
-        </CardContent>
-      </Card>
+      {isEditing ? (
+        <Card key={item._id}>
+          <CardHeader
+            action={
+              <>
+                <IconButton onClick={() => useDelete(item._id)}>
+                  <DeleteOutlineIcon />
+                </IconButton>
+                <IconButton
+                  onClick={() => setIsEditing((isEditing) => !isEditing)}
+                >
+                  <EditIcon />
+                </IconButton>
+              </>
+            }
+            title={item.name}
+          />
+          <CardContent>
+            <Typography variant="body2" color="textSecondary">
+              {/* {modifedDimension} */}
+            </Typography>
+            <Typography variant="body2" color="textSecondary">
+              {item.usage}
+            </Typography>
+            <Typography variant="body2" color="textSecondary">
+              {item.amount}
+            </Typography>
+          </CardContent>
+        </Card>
+      ) : (
+        <>
+          <Card key={item._id}>
+            <CardHeader
+              action={
+                <>
+                  <IconButton onClick={() => useDelete(item._id)}>
+                    <DeleteOutlineIcon />
+                  </IconButton>
+                  <IconButton
+                    onClick={() => setIsEditing((isEditing) => !isEditing)}
+                  >
+                    <EditIcon />
+                  </IconButton>
+                </>
+              }
+              title={item.name}
+            />
+            <CardContent>
+              <Typography variant="body2" color="textSecondary">
+                {/* {modifedDimension} */}
+              </Typography>
+              <Typography variant="body2" color="textSecondary">
+                {item.usage}
+              </Typography>
+              <Typography variant="body2" color="textSecondary">
+                {item.amount}
+              </Typography>
+            </CardContent>
+          </Card>
```

Comment:
To nie praktycznie wszystko to samo co wyżej? Dodatkowo, zdestrukturyzowałbym obiekt item wyżej zeby nie powtarzać tutaj wszedzie notacji item.prop item.name item.usage